packages/adblocker-electron-preload/src/index.ts

```diff
@@ -56,7 +59,9 @@ if (window === window.top && window.location.href.startsWith('devtools://') ===
     const scripts = getCosmeticsFiltersFirst();
     if (scripts) {
       for (const script of scripts) {
-        injectScript(script, document);
+        const result: ScriptInjectionResult = await ipcRenderer.invoke('adblocker-inject-script', script)
```

Comment:
This is a huge security risk to the library since this code may expose a chance for website to inject arbitrary script: https://www.electronjs.org/docs/latest/tutorial/context-isolation#security-considerations

Reply:
I think we shouldn't pass any script to the backend. Instead, the whole query and execution process should be done on the backend for the security.

Reply:
Thank you for catching this. I put together this patch with our security model in mind, where we set `sandbox: true`, `contextIsolation: true`, and `nodeIntegration: false`, so this wasn't a problem. I'll address the general case by moving the script execution to the main world completely.